test/k8sT/Services.go

```diff
@@ -146,4 +147,162 @@ var _ = Describe("K8sServicesTest", func() {
+		defer kubectl.Delete(bookInfoPath)
+		res.ExpectSuccess()
+
+		pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, formatLabelArgument(version, v1), helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitforPods(helpers.DefaultNamespace, formatLabelArgument(version, v2), helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		reviewsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, reviews, version, v1)).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		productpagePodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, productPage, version, v1)).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		areEndpointsReady := kubectl.CiliumEndpointWait(ciliumPodK8s1)
+		Expect(areEndpointsReady).Should(BeTrue())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", details, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", ratings, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", reviews, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		pods, err = kubectl.WaitForServiceEndpoints(helpers.DefaultNamespace, "", productPage, apiPort, helpers.HelperTimeout)
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		By("Before policy import; all pods should be able to connect")
+		shouldConnect(reviewsPodV1.String(), formatAPI(ratings, apiPort, health))
+		shouldConnect(reviewsPodV1.String(), formatAPI(ratings, apiPort, ""))
+
+		shouldConnect(productpagePodV1.String(), formatAPI(details, apiPort, health))
+		shouldConnect(productpagePodV1.String(), formatAPI(details, apiPort, ""))
+		shouldConnect(productpagePodV1.String(), formatAPI(ratings, apiPort, health))
+		shouldConnect(productpagePodV1.String(), formatAPI(ratings, apiPort, ""))
+
+		var policyPath string
+		var policyCmd string
+
+		if helpers.GetCurrentK8SEnv() != "1.6" {
+			policyPath = fmt.Sprintf("%s/policies/cnp.yaml", bookInfoPath)
+			policyCmd = "cilium policy get io.cilium.k8s-policy-name=multi-rules"
+
+		} else {
+			policyPath = fmt.Sprintf("%s/policies/cnp-deprecated.yaml", bookInfoPath)
+			policyCmd = "cilium policy get io.cilium.k8s-policy-name=multi-rules-deprecated"
+		}
+		By("Importing policy")
+		res = kubectl.Create(policyPath)
+		defer func() {
+			By("Checking that all policies were deleted in Cilium")
+			output, err := kubectl.ExecPodCmd(helpers.KubeSystemNamespace, ciliumPodK8s1, policyCmd)
+			Expect(err).Should(Not(BeNil()), "policies should be deleted from Cilium: policies found: %s", output)
+		}()
+		defer kubectl.Delete(policyPath)
```

Comment:
Be aware this is done async, so when you check if all policies were deleted in Cilium we might get test flakes on line 280 `Expect(err).Should(Not(BeNil()), "policies should be deleted from Cilium: policies found: %s", output)`

Reply:
Good point, I'll add a check for policy wait here in the next upload.

Reply:
Fixed.